api/graph/helpers.go

```diff
@@ -25,3 +28,71 @@ func handleNoRowsOnNullableType[T any, U any](result T, err error, mapper func(T
 
 	return mapper(result), nil
 }
+
+type Cursor[T any] interface {
+	Encode() (string, error)
+	Decode(string) (*Cursor[T], error)
+	Validate(input T) bool
+}
+
+type ListRecipesCursor struct {
+	Id    int
+	Limit int
+}
+
+func (cursor *ListRecipesCursor) Decode(encoded string) (*ListRecipesCursor, error) {
+	decoded, err := base64.StdEncoding.DecodeString(encoded)
+	if err != nil {
+		return nil, err
+	}
+
+	err = json.Unmarshal(decoded, cursor)
+	if err != nil {
+		return nil, err
+	}
+
+	return cursor, nil
+}
+
+func (cursor ListRecipesCursor) Encode() (string, error) {
+	str, err := json.Marshal(cursor)
+	if err != nil {
+		return "", err
+	}
+	return base64.StdEncoding.EncodeToString(str), nil
+}
+
+func (cursor ListRecipesCursor) Validate(input int) bool {
+	return cursor.Limit == input
+}
+
+type ListCommentsCursor struct {
+	PostDate time.Time
+	Limit    int
+}
```

Comment:
Same comment as above for the json tags
```suggestion

type ListCommentsCursor struct {
        PostDate time.Time `json:"post_date"`
        Limit    int       `json:"limit"`
}
```

Reply:
Same as above